src/pytesmo/validation_framework/upscaling.py

```diff
@@ -0,0 +1,391 @@
+            dataframe with "upscaled" column
+        """
+        # New upscaling methods can be specified here in the lookup table
+        up_function = {
+            "average": self._average,
+        }
+        if method not in up_function.keys():
+            raise KeyError(
+                "The selected method {} is not implemented in the upscaling options".format(method)
+            )
+        f = up_function[method]
+
+        return f(df, **kwargs)
+
+    @staticmethod
+    def _average(df, **kwargs) -> pd.Series:
+        """Simple average of each column in the dataframe"""
+        out = df.mean(axis=1)
+
+        return out
+
+    def get_upscaled_ts(
+            self,
+            gpi,
+            other_name,
+            upscaling_method="average",
+            temporal_stability=False,
+            **kwargs
+    ) -> Union[None, pd.DataFrame]:
+        """
+        Find the upscale estimate timeseries with given method, for a certain reference gpi
+
+        Parameters
+        ----------
+        gpi: int
+            gpi value of the reference point
+        other_name: str
+            name of the non-reference dataset to be upscaled
+        upscaling_method: str
+            method to use for upscaling:
+                * 'average' takes the simple mean of all timeseries
+        temporal_stability: bool, default is False
+            if True, the values are filtered using the time stability concept
+        kwargs: keyword arguments
+            argumjents for the temporal window or time stability thresholds
+
+        Returns
+        -------
+        upscaled: pd.DataFrame or None
+            upscaled time series; if there are no points under the specific gpi, None is returned
+        """
+        other_lut = self.lut[other_name]
+        # check that there are points for specific reference gpi
+        if gpi not in other_lut.keys():
+            return None
+        else:
+            other_points = other_lut[gpi]
+
+        # read non-reference points and filter out Nones
+        tss = self.get_timeseries(
+            points=other_points,
+            other_name=other_name
+        )
+        tss = [df for df in tss if df is not None]
+
+        # handle situation with single timeseries or all None
+        if len(tss) <= 1:
+            if not tss:
+                return None
+            return tss[0]
+
+        # here we collect only the variable columns; flags are irrelevant at this point and can be dropped
+        target_column = self.datasets[other_name]["columns"][0]
+        to_match = []
+        for n, point_df in enumerate(tss):
+            point_ts = point_df[target_column]
+            to_match.append(
+                point_ts.to_frame(name=target_column + "_{}".format(n))  # avoid name clashing
+            )
+
+        try:
+            tss = self.temporal_match(to_match, **kwargs)
+        # in case no match is found (result is all nans)
+        except UserWarning:
+            return None
```

Comment:
Did you test if catching a warning like this works? I thought you need to use the `warnings` module for stuff like this.